rusqlite_migration/src/errors.rs

```diff
@@ -70,6 +72,21 @@ impl From<rusqlite::Error> for Error {
     }
 }
 
+impl From<tokio_rusqlite::Error> for Error {
+    fn from(e: tokio_rusqlite::Error) -> Self {
+        match e {
+            tokio_rusqlite::Error::ConnectionClosed => Error::ConnectionClosed,
+            tokio_rusqlite::Error::Rusqlite(e) | tokio_rusqlite::Error::Close((_, e)) => {
+                Error::RusqliteError {
+                    err: e,
+                    query: Default::default(),
+                }
+            }
+            _ => unreachable!(),
```

Comment:
Since `tokio_rusqlite::Error` is non-exhaustive, we don’t actually know that this is unreachable, do we?
For instance, adding a member to the enum in a future version of tokio-rusqlite wouldn’t be a breaking change (because the enum is marked `#[non_exhaustive]`) and we would end up in this branch of the match.
Maybe it would make sense to have something like `Error::Unrecognized(Box<std::Error>)` to generically bubble up errors that are not currently recognized but might be introduced by future versions of our dependencies. The benefit of a pointer like this is that we don’t have to specify the type and we can use it for all unsupported errors.

Reply:
Good idea, thanks!